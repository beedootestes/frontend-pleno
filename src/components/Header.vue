<template>
  <section class="header">
    <BarHeader class="header-bar" />
    <SearchBar />
  </section>
</template>

<script setup lang="ts">
import BarHeader from "./BarHeader.vue";
import SearchBar from "./SearchBar.vue";
</script>

<style scoped lang="scss">
  .header {
    display: flex;
    width: 100%;
    flex-direction: column;
    border-bottom: 1px solid $color-border;
    box-shadow: 0 15px 10px -15px $shadow;
    box-sizing: border-box;
    padding: 20px 25px 15px 25px;

    .header-bar {
      margin-bottom: 15px;
    }
  }
</style>