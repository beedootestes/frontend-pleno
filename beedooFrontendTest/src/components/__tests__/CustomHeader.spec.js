import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import Header from '../CustomHeader.vue'

describe('Testa se o componente Header é uma instância do Vue', () => {
  it('is a Vue instance', () => {
    const wrapper = mount(Header)
    expect(wrapper.vm).toBeTruthy()
  })

  it('renderiza corretamente', () => {
    const wrapper = mount(Header)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
