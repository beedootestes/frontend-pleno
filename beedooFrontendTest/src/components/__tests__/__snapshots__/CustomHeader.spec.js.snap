// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testa se o componente Header é uma instância do Vue > renderiza corretamente 1`] = `
"<v-container data-v-8942cb99=\\"\\">
  <v-card data-v-8942cb99=\\"\\" class=\\"d-flex flex-column card elevation-5 grey-lighten-3\\">
    <v-row data-v-8942cb99=\\"\\">
      <v-col data-v-8942cb99=\\"\\" class=\\"ma-2\\">
        <v-img data-v-8942cb99=\\"\\" src=\\"../../public/logo-beedoo.png\\" alt=\\"logo-beedoo\\" width=\\"200\\" height=\\"50\\"></v-img>
      </v-col>
      <v-col data-v-8942cb99=\\"\\" class=\\"d-flex flex-row-reverse\\">
        <v-btn data-v-8942cb99=\\"\\" class=\\"ma-2\\" color=\\"grey-lighten-2\\" icon=\\"mdi-account-circle\\"></v-btn>
        <v-btn data-v-8942cb99=\\"\\" class=\\"ma-2\\" color=\\"grey-lighten-2\\" icon=\\"mdi-cart\\"></v-btn>
      </v-col>
    </v-row>
    <v-row data-v-8942cb99=\\"\\" class=\\"ml-2 mr-2 mb-5\\">
      <v-text-field data-v-8942cb99=\\"\\" density=\\"compact\\" variant=\\"solo\\" label=\\"Pergunte para o beebot\\" append-inner-icon=\\"mdi-microphone\\" single-line=\\"\\" hide-details=\\"\\"></v-text-field>
    </v-row>
  </v-card>
</v-container>"
`;
